### ISSUE TITLE
[TF2] "Out of memory or address space" on Linux w/Intel drivers

### ISSUE THREAD

**Issue by nightsky30** [ValveSoftware/Source-1-Games#2212]
I'm receiving this error randomly while playing TF2.  The game crashes.

"Out of memory or address space.  Texture quality setting may be too high."

I don't think this is acceptable.  I have a 64bit Core i7 system with 16GB of RAM.  That should be far more than enough RAM for a game which debuted October 9th, 2007.

Valve lists the following specs (SteamOS + Linux) on their TF2 store page:

MINIMUM:
OS: Ubuntu 12.04
Processor: Dual core from Intel or AMD at 2.8 GHz
Memory: 1 GB RAM
Graphics: nVidia GeForce 8600/9600GT, ATI/AMD Radeon HD2600/3600 (Graphic Drivers: nVidia 310, AMD 12.11), OpenGL 2.1
Network: Broadband Internet connection
Storage: 15 GB available space
Sound Card: OpenAL Compatible Sound Card
Additional Notes: Mouse, Keyboard

1GB OF RAM...Why is this an issue.  The user should not be made to think they must reduce their texture quality settings.  Someone, Valve/xorg/intel, should fix their memory leaks.

My system information:

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Laptop
    No Touch Input Detected
    
Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
    CPU Family:  0x6
    CPU Model:  0x8e
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  3500 Mhz
    4 logical processors
    2 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    Ubuntu 16.04.2 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.8.0-42-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11804000
    X Window Manager:  Compiz
    Steam Runtime Version:  steam-runtime-beta-release_2016-12-05
    
Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) Kabylake GT2 x86/MMX/SSE2

    Driver Version:  3.0 Mesa 12.0.6
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x8086
    DeviceID:  0x5916
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 11.57" x 6.50"  (13.27" diag)
                                            29.4cm x 16.5cm  (33.7cm diag)
    Primary VRAM Not Detected
    
Sound card:
    Audio device: Realtek ALC3246
    
Memory:
    RAM:  15932 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Steam Controller Cable and Base:  Not set
    Total Hard Disk Space Available:  445049 Mb
    Largest Free Hard Disk Block:  49247 Mb
    VR Headset: None detected
    
Recent Failure Reports:
    Wed Mar 22 23:45:33 2017 GMT: file ''/tmp/dumps/crash_20170322183525_1.dmp'', upload yes: ''CrashID=bp-c6cd4a36-aeac-420e-8639-b77d22170322''

**Comment by nightsky30** [comment]
Here is a steam thread referencing a similar, if not the same issue back in 2016:

http://forums.steampowered.com/forums/showthread.php?t=3301320

**Comment by kisak-valve** [comment]
Hello @nightsky30, mesa/Intel uses per-process system ram for graphics operations, which makes it easier to hit the 4GB limit of 32 bit applications compared to systems with a discrete GPUs. This is possibly why Intel graphics is not included as officially supported in the minimum specs.

**Comment by Wasmanr2** [comment]
Delete sound.cache in tf2.
This is the same issue i've posted with a temp fix.
This temp fix should work regardless of linux,windows ubuntu etc.


>From issue #2187  
(Source: https://facepunch.com/showthread.php?t=1528873)

A temporary fix for this is deleting the sound.cache before starting tf2 or while being ingame deleting sound.cache and writing snd_restart in the console.

**Comment by jljusten** [comment]
@nightsky30:
* Did it start happening after a recent TF2 update?
* How easy is to reproduce?
* What steps can be used to reproduce the crash?
* Does lowering the quality settings help?
* Does the game give the error message and exit, or it actually crashes and stops running without notifying you?

**Comment by nightsky30** [comment]
@jljusten 

- I'm not aware of any recent update that triggered this behavior.  I just started playing this game again after a long period of time, and I've not played it on this system before.
- I've not attempted reproduction.  This was the first occurrence I've ever witnessed.
- I just played the game for maybe 20 minutes on a community server playing 2fort with 0 mods
- I have not lowered the quality settings
- The game displays the error message as it freezes game-play.  Upon clicking "Ok", TF2's processes are terminated.  There is a crash report listed above in my system information which corresponds with the approximate time this error was witnessed.

**Comment by nightsky30** [comment]
@Wasmanr2 

Interesting.  I'd done a search for "Out of mem" under open issues before posting this.  Maybe I should have used a longer search string.  Thanks for catching this.  It might be the same issue :)  I've not looked into console output yet.

**Comment by nightsky30** [comment]
@kisak-valve 

Thank you for the info.  I was not aware TF2 was running as a 32bit process.  I thought perhaps it was also ported/available as 64bit to those who have capable systems.  That being said, I still think memory management should be improved to prevent this situation from happening.

If it is the same sound related issue Wasmanr2 reported, the error message is misleading.

**Comment by hahagoni** [comment]
Although it's not likely it will help, but you should try it with the newest Mesa release
[link](https://www.mesa3d.org/relnotes/13.0.6.html)

**Comment by nightsky30** [comment]
@hahagoni 

I'm running Ubuntu 16.04.2 LTS w/ HWE (Hardware Enablement) stack enabled to take advantage of the newer kernel and video drivers as my system is newer.  I tried moving from the hwe to hwe-edge which is the farthest I can go while still maintaining at 16.04 LTS.

HWE-Edge allowed me to move from the 4.8 kernel to the 4.10 kernel, but apparently there is no available hwe-edge xorg package, and there are not any newer mesa drivers available in the xenial repo than 12.06.

I'm not willing to upgrade from 16.04 LTS yet, and I'd rather not manually install something as important as mesa drivers.

Thank you for the suggestion though :)

**Comment by nightsky30** [comment]
Updated system information:

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Laptop
    No Touch Input Detected
    
Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
    CPU Family:  0x6
    CPU Model:  0x8e
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  3500 Mhz
    4 logical processors
    2 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    Ubuntu 16.04.2 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.10.0-13-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11804000
    X Window Manager:  Compiz
    Steam Runtime Version:  steam-runtime-beta-release_2016-12-05
    
Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) Kabylake GT2 x86/MMX/SSE2

    Driver Version:  3.0 Mesa 12.0.6
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x8086
    DeviceID:  0x5916
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 11.57" x 6.50"  (13.27" diag)
                                            29.4cm x 16.5cm  (33.7cm diag)
    Primary VRAM Not Detected
    
Sound card:
    Audio device: Realtek ALC3246
    
Memory:
    RAM:  15929 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Steam Controller Cable and Base:  Not set
    Total Hard Disk Space Available:  445049 Mb
    Largest Free Hard Disk Block:  48826 Mb
    VR Headset: None detected
    
Recent Failure Reports:
    Thu Mar 23 17:10:40 2017 GMT: file ''/tmp/dumps/assert_20170323121039_23.dmp'', upload yes: ''CrashID=bp-7e11d451-63a2-44a3-b51d-ee4f12170323''

**Comment by kisak-valve** [comment]
If you want to try out a newer version of mesa on Ubuntu, [Padoka's Stable Mesa PPA](https://launchpad.net/~paulo-miguel-dias/+archive/ubuntu/pkppa) is a relatively safe option.

**Comment by nightsky30** [comment]
I have tried the PPA with 17.0.1 Mesa packages as suggested.  I played TF2 for a while, and have received the same error.

**Comment by Nephyrin** [comment]
Please note that deleting sound cache files will result in significant FPS stuttering due to synchronous I/O being performed to compensate for cache misses.  If you have deleted these files, you should consider running "verify integrity of game files" in Steam game properties to restore them.

As for the memory issues on intel drivers -

**tl;dr**: Ultimately, memory management improvements in the drivers, or a full 64bit port of the game, will likely be required to play on high texture quality settings for these cards.

*Longer version*: TF can load a great deal of textures on high texture quality settings. When drivers opt to store these textures, occasionally multiple times, in process memory, they can quickly dwarf the memory used by the engine itself and lead to exhaustion of the virtual address space. Setting texture quality to a lower value in video options will help avoid this.

The engine currently contains various workarounds for pathological driver cases, but ultimately the drivers in question simply do not operate on 1GB+ of textures in a 32bit process efficiently.  The game will often hit the exhaustion limit having used less than half of virtual memory, with the rest being driver allocations.

**Comment by Wasmanr2** [comment]
I know @Nephyrin but those 5 mins or longer of building up missing sound cache are worth to wait when the crash is fixed with that option.



Verifing integerity of game files brings mostly nothing if texes or models or sounds aren't missing.(Just saying i might be wrong but it never helped that much).
(And ofc i know you guys have a whole lot of experience and stuff since you guys are working on it and not me).

**Comment by Wasmanr2** [comment]
@nightsky30 did the fix of deleting sound.cache worked for you?

**Comment by nightsky30** [comment]
I've actually not tried that.  It sounds as if it will just cause lag and eat bandwidth if the true problem is in the memory management of the game and intel drivers.

I'd rather Valve, 01.org, and freedesktop.org work together to fix the issues

**Comment by Wasmanr2** [comment]
Alright your decision but deleting cache wont eat bandwidth.
Just sayin because a cache is a temporary storage generated by the engine to locate files faster (if im right).

**Comment by bigstiffy** [comment]
I've had this same issue now for months, and on certain servers the crash happens within a minute or two of joining. I have an nvidia graphics card (2gb VRAM), 16gb of RAM, and I have tf2 installed on a solid state drive. Originally I thought maybe it was an issue tied to my graphics card, which I monitored through commandline with nvidia-smi, and watched as the card's memory was gradually maxed out.
However, a crash would occasionally happen before the card's memory was full, ruling out the card being at fault. So I ran `top` instead, and saw that tf2 rapidly reached a 4gb limit and promptly crashed.
This is all post-resetting both steam and tf2.

**Comment by s0r00t** [comment]
Been having this issue for more than a year (#1981), on a "worse" computer. Arch Linux x64, Intel GPU. The game hangs at the initial loading screen, then it eithers shows the error or just freeze.
It started back in the Gun Mettle update, but deleting sound.cache fixed it in the past.  
Since Tough Break, it doesn't work anymore.

While this isn't great for us players, how can we "downgrade" textures without being able to run the game?

**Comment by kisak-valve** [comment]
I'm not sure it will help in this case, but you can run TF2 once with `-autoconfig` in the launch options to reset the config to the default. Remember to remove it afterwards so that changes you make are retained.

**Comment by nightsky30** [comment]
In Man vs Machine, the game crasdhes almost immediately when waves begin.  I think the amount of bot models spawning may be the culprit.  Might fill up the memory even faster.

**Comment by Prinzhorn** [comment]
Same here. Fresh Ubuntu 17.04 with fresh Steam install. New Intel NUC Skull Canyon with 16GB RAM and 250GB SSD. Just played my first game of TF2 and it crashed after a few minutes.

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected
    
Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-6770HQ CPU @ 2.60GHz
    CPU Family:  0x6
    CPU Model:  0x5e
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  3500 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    Ubuntu 17.04 (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.10.0-21-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11903000
    X Window Manager:  Compiz
    Steam Runtime Version:  steam-runtime-beta-release_2016-12-05
    
Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) Iris Pro Graphics 580 (Skylake GT4e) x86/MMX/SSE2

    Driver Version:  3.0 Mesa 17.0.3
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x8086
    DeviceID:  0x193b
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  2560 x 1440
    Desktop Resolution: 2560 x 1440
    Primary Display Size: 21,77" x 12,24"  (24,96" diag)
                                            55,3cm x 31,1cm  (63,4cm diag)
    Primary VRAM Not Detected
    
Sound card:
    Audio device: Realtek ALC233
    
Memory:
    RAM:  15934 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Steam Controller Cable and Base:  Not set
    Total Hard Disk Space Available:  233201 Mb
    Largest Free Hard Disk Block:  118428 Mb
    VR Headset: None detected
    
Recent Failure Reports:
    Fri Jun  2 14:30:19 2017 GMT: file ''/tmp/dumps/crash_20170602155455_1.dmp'', upload yes: ''CrashID=bp-2a5c29a1-8474-4c18-8728-a2c802170602''

**Comment by notchristopher** [comment]
I am currently facing a similar issue with Team Fortress 2. It's not the "Out of memory" crash, but the game slowly eats RAM until ~2.7GB, from startup to the main menu, and 3GB (!) during casual play. Meanwhile, it doesn't go past ~1.1GB usage on Windows, even with community servers.
This issue has been following me since my first GNU/Linux experience (more than two years ago).

Distro: Void Linux
Desktop: MATE 1.18.0
Laptop: Dell Latitude E5470

```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Laptop
    No Touch Input Detected
    
Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
    CPU Family:  0x6
    CPU Model:  0x4e
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  3000 Mhz
    4 logical processors
    2 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    "Void Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.11.5_1
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11903000
    X Window Manager:  Metacity (Marco)
    Steam Runtime Version:  steam-runtime-beta-release_2016-12-05
    
Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) HD Graphics 520 (Skylake GT2) x86/MMX/SSE2

    Driver Version:  3.0 Mesa 17.1.2
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x8086
    DeviceID:  0x1916
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1366 x 768
    Desktop Resolution: 1366 x 768
    Primary Display Size: 12.20" x 6.69"  (13.90" diag)
                                            31.0cm x 17.0cm  (35.3cm diag)
    Primary VRAM Not Detected
    
Sound card:
    Audio device: Realtek ALC3235
    
Memory:
    RAM:  3903 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Steam Controller Cable and Base:  Not set
    Total Hard Disk Space Available:  336926 Mb
    Largest Free Hard Disk Block:  254729 Mb
    VR Headset: None detected
    
Recent Failure Reports:
    

```

Any recommendations to at least minimize this issue? Thanks.

**Comment by ibrokemypie** [comment]
Void linux, 16gb ram, intel 4770k, I assumed this was vram, seems that tf2 doesnt ever clear the vram, so after playing multiple maps they all get cached, very quickly leading to oom with the only like 256mb vram of an igpu

**Comment by nightsky30** [comment]
Is there any chance TF2 will be ported to the Source 2 engine now that it is out?  I'm hoping there is a possibility so Intel GPUs might benefit.  It seems DOTA2 and CS:GO are getting ported.  TF2 may not be the most popular game compared to the other aforementioned titles, but the community is still rather large, and I would guess Valve has made a lot of money on that game.

**Comment by ibrokemypie** [comment]
ugh
like every two maps it crashes, and in casual you are then forced to rejoin a game, even if the game is now empty

**Comment by ibrokemypie** [comment]
```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz
    CPU Family:  0x6
    CPU Model:  0x3c
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  4000 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Void Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.11.8_1
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11903000
    X Window Manager:  i3
    Steam Runtime Version:  steam-runtime-beta-release_2016-12-05

Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) Haswell Desktop x86/MMX/SSE2
    Driver Version:  3.0 Mesa 17.1.4
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x8086
    DeviceID:  0x412
    Revision Not Detected
    Number of Monitors:  2
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 3520 x 1200
    Primary Display Size: 20.87" x 11.81" (23.98" diag)
                                            53.0cm x 30.0cm (60.9cm diag)
    Primary VRAM Not Detected

Sound card:
    Audio device: Intel Haswell HDMI

Memory:
    RAM:  15716 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_AU.UTF-8
    Total Hard Disk Space Available:  937875 Mb
    Largest Free Hard Disk Block:  251301 Mb
    VR Headset: None detected

Recent Failure Reports:
```

**Comment by nightsky30** [comment]
Saw this [article](https://www.phoronix.com/scan.php?page=news_item&px=Mesa-Flush-Control-Lands), and was wondering if this future driver feature slated for Mesa 17.4/18.0 might help with this issue...

**Comment by ibrokemypie** [comment]
Am now getting the same error on an AMD RX Vega 56...

**Comment by AaronBottegal** [comment]
I hit this bug on a Ryzen system with a GTX 1070 8GB VRAM GPU. I have 32GB CPU RAM. It used about 8GB of it then crash, didn't realize until it just came up and crashed. Only experienced it once ever, today.

**Comment by nightsky30** [comment]
I hope this fixes the issue:
https://cgit.freedesktop.org/mesa/mesa/commit/?id=cee9f3890351e24cfaa000f32fef10367c45fe64

**Comment by Technohacker** [comment]
@nightsky30 That'd fix it for the i965 chipset. However, for people like me on the i915 chipset, there's no change :\

**Comment by nightsky30** [comment]
@Technohacker Let's hope they patch the i915/945 for consistency.

I'd like to test this on the i965, but 2 things stand in the way:

1:  The current Padoka PPA build that is based on master does not yet include this commit.  The next build should.
2:  Immediate crashes are still occurring due to whatever bug is causing #2310 and #2435 I reported it here:  https://bugs.freedesktop.org/show_bug.cgi?id=103810  The exact cause is unknown.  I cannot test the memory issue if another bug crashes the game constantly.  There's no time for memory to be used up by normal game play.

**Comment by kisak-valve** [comment]
There's some misunderstanding going on here, This issue is with TF2 being killed from running out of virtual memory address space for the entire application. Intel and other iGPUs use part of the 4GB limit inherent with all 32 bit applications.

Historically, many graphics drivers have gotten named based on the oldest chipset supported by a driver. In this case, the kernel module for Intel is called i915, and in userspace is can be called Mesa/i965. There also is Mesa/i915. Mesa/i965 is Intel generation 4 graphics, which is the oldest Intel chipset capable of OpenGL 2.1. While theoretically possible, Steam does not render properly on pre-OpenGL 2.1 hardware and Team Fortress 2 requires OpenGL 2.1 + sRGB. Team Fortress 2 will not work on Mesa/i915 regardless.

The purpose of the Intel generation 8+ patch in mesa is to potentially expose as much physical system memory to mesa as possible to potentially be used by the GPU, and does not influence the 4GB virtual memory limit of 32 bit applications.

**Comment by Technohacker** [comment]
@kisak-valve I tried looking around for the actual codename for my Intel HD Graphics 3000, couldn't find it. But (wild guess), do you think [this patch](https://lists.freedesktop.org/archives/mesa-dev/2017-October/173828.html) would be of use for reduction of memory use?

**Comment by Technohacker** [comment]
So wait, the 2nd gen Intel HD Graphics 3000 I have on my laptop is supposed to be pre-OpenGL 2.1? Yet it seems to me I have OpenGL 3.x support (in use by KWin in KDE):

`OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) Sandybridge Mobile 
OpenGL core profile version string: 3.3 (Core Profile) Mesa 17.3.6`

**Comment by kisak-valve** [comment]
Sandybridge is second generation "Intel HD", but overall is [Intel generation 6 graphics](https://en.wikipedia.org/wiki/List_of_Intel_graphics_processing_units#Sixth_generation).

**Comment by Technohacker** [comment]
@kisak-valve Ah, yes. I just noticed it in Wikipedia, sorry for the bother :sweat_smile:

**Comment by robsco-git** [comment]
Would running TF2 through a WINE Steam install mitigate this issue?

**Comment by kisak-valve** [comment]
One of the [changes](https://gitlab.freedesktop.org/mesa/mesa/commit/4066610ad4956ff6bb0afefb767d7577b72d6081) in mesa 18.1.5 potentially improved TF2's behavior on mesa/Intel.

Can someone check if mesa 18.1.5 makes this issue less likely?

**Comment by psposito** [comment]
Hi,

I have installed Pirate Vikings and Knight II on my Arch Linux (Antergos) with Nvidia 950 GTX, mesa 18.1.5 and I am getting the same message.

[https://i.imgur.com/dronrJBl.png](url)

**Comment by psposito** [comment]
@kisak-valve 
I was redirected here....

**Comment by Technohacker** [comment]
@kisak-valve TF2 works on Intel Graphics on Arch Linux. Mesa 18.1.5.

**Comment by agret** [comment]
https://www.phoronix.com/scan.php?page=news_item&px=Mesa-19.0-Halves-TF2-Memory

**Comment by Nephyrin** [comment]
This should be greatly improved as of today's update, and further improved by Mesa 19.0.

**Comment by Nephyrin** [comment]
We're not aware of any configurations suffering from memory issues on Linux after the recent changes, but please re-open if you're still seeing this error.

**Comment by radrow** [comment]
Hi, after my recent full system update this problem appeared for me (during Halloween theme everything was ok). Clearing sound cache didn't help. 

My `uname`:
```
Linux HPOmen 5.3.10-arch1-1 #1 SMP PREEMPT Sun, 10 Nov 2019 11:29:38 +0000 x86_64 GNU/Linux
```
Other random stuff:
```
OS: Arch Linux
Kernel: x86_64 Linux 5.3.10-arch1-1
CPU: Intel Core i7-7700HQ @ 8x 3.8GHz
GPU: GeForce GTX 1050
RAM: 2559MiB / 15893MiB
```
Anything else I can provide?

**Comment by bigstiffy** [comment]
I am now getting the same error as radrow, when just a week ago it was capable of running.
Application encounters error on startup before main menu can load, upon clicking OK on the error it takes about a minute to crash fully.
Also running Arch Linux and recently updated my system, when I haven't gotten this error in over a year.

I currently have these packages installed:
mesa 19.2.4-1
glu 9.0.1-1
lib32-mesa 19.2.4-1
lib32-glu 9.0.1-1
and using proprietary NVIDIA drivers

What has not fixed it:
*Setting startup options that would lower settings (_GLVND_DISALLOW_PATCHING=1 SDL_AUDIODRIVER=pulseaudio %command% -novid -autoconfig -32bit -safe -gl_enablesamplerobjects -full_memory_dumps), but it only changed the reliability of getting the engine error, which otherwise might not even show and TF2 would simply crash and systemd would perform a core dump.
*Verifying integrity of game files.
*Deleting TF2 cache folder
*Starting steam with native libraries instead of steam's, made no difference.
*Disabling steam overlay.
*Opting into pre-release update (no update was scheduled so I assume it's older).
*Opting into pre_jungleinferno_demos

**Comment by kisak-valve** [comment]
Hello @radrow, @bigstiffy, you may have some luck trying https://github.com/ValveSoftware/Source-1-Games/issues/2876#issuecomment-553748774.

**Comment by bigstiffy** [comment]
@kisak-valve
Tested and confirmed that does fix the crashing on startup. Thank you for linking that comment, I probably would not have seen it otherwise.

**Comment by radrow** [comment]
Helped me as well
